### PR TITLE
Add paper contribution docs for events

### PR DESCRIPTION
**Pull request by Machine-Maker**

### CODE REVIEW

File: docs/paper/contributing/events.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+---
+slug: /contributing/events
+---
+
+# Events
+There are several requirements for events in the Paper API. Note that while not all existing events may follow these
```
Comment (olijeffers0n) [inline docs/paper/contributing/events.md:6]: this is the bit that shows up in the embed. Maybe make a slightly more user friendly sentence for this, move this down and the note could even go in an amondition

File: docs/paper/contributing/events.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+guidelines, all new and modified events should adhere to them.
+
+* New events should go in the package (or sub-package of) `io.papermc.paper.event`. 
+* A `HandlerList` field in an event should be private and static and have the proper capitalization (HANDLER_LIST)
+* New constructors should be annotated with `ApiStatus.Internal` to further signify that they are not considered API.
+* Constructors that are being replaced, if they aren't being removed, should be marked as `Deprecated` and `DoNotUse`.
+* [Mutability](#mutability)
```
Comment (olijeffers0n) [inline docs/paper/contributing/events.md:13]: Maybe requires more than one word. this looks off

File: docs/paper/contributing/events.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+### Mutability
+Certain API types are "mutable" which can lead to unexpected behavior within events. Mutable types like
+`Location` and `Vector` should therefore be cloned when returned from a "getter" in an event.
+
+
+### HandlerList
+An event type is only valid for event-handler registration if that type or a super-type has the `HandlerList` field and methods on it.
+Consider not putting a `HandlerList` on every event, just a "common parent" event so that a plugin can listen to the 
+parent event and capture any child events but also listen to the child event separately.
+
```
Comment (olijeffers0n) [inline docs/paper/contributing/events.md:26]: Extra newline

File: docs/paper/contributing/events.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+There are several requirements for events in the Paper API. Note that while not all existing events may follow these
+guidelines, all new and modified events should adhere to them.
+
+* New events should go in the package (or sub-package of) `io.papermc.paper.event`. 
+* A `HandlerList` field in an event should be private and static and have the proper capitalization (HANDLER_LIST)
+* New constructors should be annotated with `ApiStatus.Internal` to further signify that they are not considered API.
+* Constructors that are being replaced, if they aren't being removed, should be marked as `Deprecated` and `DoNotUse`.
+* [Mutability](#mutability)
+* All `ItemStack` parameters should be marked as `@NotNull`.
+
```
Comment (olijeffers0n) [inline docs/paper/contributing/events.md:15]: Why double new lines?

**Review by olijeffers0n — CHANGES_REQUESTED** [review]
Few Changes

**Comment by olijeffers0n** [comment]
Your .editorconfig has made it in somehow, also need to look at the redirect to check it. Otherwise the content looks generally good

**Comment by Machine-Maker** [comment]
oh editorconfig should be in, but I guess it can be a different PR.

**Review by olijeffers0n — APPROVED** [review]
I’ve looked over the sidebar config and all looks good. Content of the page is good for now and can be added to in the future as needed. Up to you, but i’m happy for this editorconfig to stay here

**Review by NotMyFault — APPROVED** [review]

**Pull request merged**